File: src/xPortals.ServiceModel/DTOs/Leasing/LeaseTermination.cs
using ServiceStack;
using System;

namespace xPortals.DTOs.Leasing
{
    public class LeaseTermination : IReturn<LeaseTerminationResponse>
    {
        public string Type { get; set; }
        public Guid? Id { get; set; }
    }

    public class LeaseTerminationResponse
    {
        public DateTime LeaseExpiryDate { get; set; }
        public ResponseStatus ResponseStatus { get; set; }
    }

    public class LeaseTerminationPost : IReturn<LeaseTerminationPostResponse>
    {
        public string Type { get; set; }
        public Guid? Id { get; set; }
        public string Reason { get; set; }
        public string OtherReason { get; set; }
        public DateTime TerminationDate { get; set; }
    }

    public class LeaseTerminationPostResponse
    {
        public DateTime LeaseExpiryDate { get; set; }
        public ResponseStatus ResponseStatus { get; set; }
    }
}

File: src/xPortals/ClientApp/xPortals/Leasing/leaseTermination.ts
import {
    LeaseTermination as Request, LeaseTerminationResponse as Response,
    LeaseTerminationPost as PostRequest, LeaseTerminationPostResponse as PostResponse
} from "../../xPortalsApi.dtos";
import { autoinject } from 'aurelia-framework';
import { client } from "../shared";
import { DateValueConverter } from "../resources/value-converters/date";

@autoinject
export class LeaseTermination {
    isLoading: boolean = false;

    entity = new Request();
    result: Response;
    postRequest = new PostRequest();
    postResponse: PostResponse;

    constructor(private date: DateValueConverter) { }

    activate(params: any) {
        this.entity.type = this.postRequest.type = params.type;
        this.entity.id = this.postRequest.id = params.id;
        return client.get(this.entity).then(result => {
            this.result = result;
            this.postRequest.terminationDate = this.date.toView(this.result.leaseExpiryDate, 'YYYY-MM-DD');
        }).catch(result => {
            console.error(result);
            this.result = result;
        });
    }

    submit() {
        client.post(this.postRequest).then(result => {
            this.postResponse = result;
        }).catch(result => {
            console.error(result);
            this.postResponse = result;
        });
    }
}